app/models/peep.rb

```diff
@@ -0,0 +1,14 @@
+
+class Peep
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :text, Text, required: true
+  property :created_at, DateTime
+
+  has 1, :user, through: Resource
+
+  validates_presence_of :text
```

Comment:
Prefer the new style validations `validates :column, presence: value` over `validates_presence_of`.